packages/dd-trace/src/ci-visibility/dynamic-instrumentation/index.js

```diff
@@ -0,0 +1,92 @@
+'use strict'
+
+const { join } = require('path')
+const { Worker } = require('worker_threads')
+const { randomUUID } = require('crypto')
+const log = require('../../log')
+const { BREAKPOINT_HIT, BREAKPOINT_SET } = require('./messages')
+
+const probeIdToResolveBreakpointSet = new Map()
+const probeIdToResolveBreakpointHit = new Map()
+
+class TestVisDynamicInstrumentation {
+  constructor () {
+    this.worker = null
+    this._readyPromise = new Promise(resolve => {
+      this._onReady = resolve
+    })
+  }
+
+  // Return 3 elements:
+  // 1. Snapshot ID
+  // 2. Promise that's resolved when the breakpoint is set
+  // 3. Promise that's resolved when the breakpoint is hit
+  addLineProbe ({ file, line }) {
+    const snapshotId = randomUUID()
+    const probeId = randomUUID()
+
+    this.worker.postMessage({
+      snapshotId,
+      probe: { id: probeId, file, line }
+    })
+
+    return [
+      snapshotId,
+      new Promise(resolve => {
+        probeIdToResolveBreakpointSet.set(probeId, resolve)
+      }),
+      new Promise(resolve => {
+        probeIdToResolveBreakpointHit.set(probeId, resolve)
+      })
+    ]
+  }
+
+  isReady () {
+    return this._readyPromise
```

Comment:
this is just to make sure we don't attempt to set probes before the worker is online